govstack_api/building_blocks/bb_digital_registries/registries/insuree_registry.py

```diff
@@ -211,38 +118,34 @@ def extract_uuid(self, mapped_data_query):
         else:
             return None
 
-    def dict_to_xml(data_dict):
-        xml_elements = ET.Element('root')
-        for key, value in data_dict.items():
-            child = ET.SubElement(xml_elements, key)
-            child.text = str(value)
-        return minidom.parseString(ET.tostring(xml_elements)).toprettyxml()
-
-    def change_result_extension(self, data, extension):
-        if extension == 'json':
-            return json.dumps(data)
-        elif extension == 'string':
-            return str(data)
-        elif extension == 'xml':
-            return self.dict_to_xml(data)
-        else:
-            raise ValueError(f"Unknown extension type: {extension}")
-
-    def encode_id(self, mapped_data):
-        if 'id' in mapped_data:
-            id_value = mapped_data['id']
-            mapped_data['id'] = base64.b64encode(f"Insuree:{id_value}".encode()).decode()
-        return mapped_data
-
-    def create_arguments_with_values(self, mapped_data) -> str:
+    def insuree_encode_id(self, mapped_data):
+        return self.encode_id(mapped_data, "Insuree")
+
+    def get_required_data_for_mutation(self, mapped_data: dict) -> dict:
+        """
+        This function adapts the input data to the specific schema of the registry.
+        """
+        adapted_data = self.default_values.copy()
+        adapted_data.update(mapped_data)
+        if 'chfId' not in adapted_data and 'id' in mapped_data:
+            adapted_data['chfId'] = f'chfId: "{mapped_data["id"]}"'
+        for key in mapped_data.keys():
+            if key in adapted_data:
+                del adapted_data[key]
+        return adapted_data
+
+    def create_arguments_with_values(self, mapped_data: dict) -> str:
         arguments = []
         for field, value in mapped_data.items():
             if field == 'jsonExt' and isinstance(value, str):
                 formatted_value = value.replace('"', '\\"')
                 arguments.append(f'{field}: "{formatted_value}"')
+            elif field == 'id':
+                try:
+                    value = int(value)
+                    arguments.append(f'{field}: {value}')
+                except ValueError:
+                    arguments.append(f'{field}: "{value}"')
```

Comment:
This part is kind of workaround, if I get this right it's due to ambiguity of primary keys im openIMIS. I think you should left a comment why given logic is in here and also I think that this code could be refactored. I'm not a fan of try/catch block as it could lead to some unexpected results. For instance: floating points, negative integers, non-string inputs could be valid from perspective of this block but not overall logic. You can use `str.isdigit()` check to determine if legacy ID is in use. Please note that both solutions have pros and cons and in other context the more robust try/catch block could be a better option.

Reply:
Thanks, I will use `str.isdigit()` and I will add comments with explanation